cdisc_rules_engine/operations/record_count.py

```diff
@@ -14,5 +14,18 @@ def _execute_operation(self) -> pd.Series:
         4    5
         dtype: int64
         """
-        record_count: int = len(self.params.dataframe)
-        return record_count
+        filtered = None
+        result = len(self.params.dataframe)
+        if self.params.filter:
+            filtered = self._filter_data(self.params.dataframe)
+            result = len(filtered)
+        if self.params.grouping:
+            self.params.target = "size"
+            if filtered is not None:
+                group_df = filtered.groupby(self.params.grouping, as_index=False).size()
+            else:
+                group_df = self.params.dataframe.groupby(
+                    self.params.grouping, as_index=False
+                ).size()
```

Comment:
I think this can be further simplified to:
```python
            group_df = (
                (filtered or self.params.dataframe)
                .groupby(self.params.grouping, as_index=False)
                .size()
            )
```

Reply:
I wanted to do something like this but unfortunately, pandas doesn't allow truthiness testing of a dataframe so I ended with an if/else clause